syft/frameworks/crypten/__init__.py

```diff
@@ -1,17 +1,26 @@
 import torch
 import syft
+
 from syft.frameworks.crypten.context import toy_func, run_party
+from syft.exceptions import WorkerNotFoundException
+
 import crypten.communicator as comm
 import crypten
 
 
-def load(tag: str, src: int, id_worker: int):
-    if comm.get().get_rank() == src:
-        worker = syft.local_worker.get_worker(id_worker)
-        result = worker.search(tag)[0].get()
+def load(tag: str, src: int):
+    worker = syft.local_worker.get_worker_from_rank(src)
+
+    if worker and src == worker.rank:
```

Comment:
Can't we simply check either `src == comm.get().get_rank()` ?

Reply:
Yep...you are right. I overcomplicated this :(